Shellcheck shows warnings for the wrapper shell script

### Current Behavior

The `gradlew` wrapper scripts for Gradle 8.3 (and the current 8.4 snapshot version of Gradle) contain warnings detected by Shellcheck integration with IntelliJ IDEA (via bundled Shell Script plugin).

![image](https://github.com/gradle/gradle/assets/2759152/a449d14a-7d46-4dd4-924e-741958e52245)


### Expected Behavior

No shell script warnings are reported

### Context (optional)

This is annoying also because IDEA detects those warnings as a part of the commit checks and points the user to them. The users are not supposed to edit the wrapper scripts, so there is no good way of fixing that rather than disabling the Shellcheck, which is most likely not desired.

One check is already ignored for those lines, and I guess we need to add another ignore for the https://github.com/koalaman/shellcheck/wiki/SC2039

### Steps to Reproduce

Generate a any project with `gradle init` and run Shellcheck on the `gradlew`

### Gradle version

8.4 master

### Build scan URL (optional)

_No response_

### Your Environment (optional)

_No response_

--- a/subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt
+++ b/subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt
@@ -180,7 +180,7 @@ if ! "\$cygwin" && ! "\$darwin" && ! "\$nonstop" ; then
     case \$MAX_FD in #(
       max*)
         # In POSIX sh, ulimit -H is undefined. That's why the result is checked to see if it worked.
-        # shellcheck disable=SC3045
+        # shellcheck disable=SC2039,SC3045
         MAX_FD=\$( ulimit -H -n ) ||
             warn "Could not query maximum file descriptor limit"
     esac
@@ -188,7 +188,7 @@ if ! "\$cygwin" && ! "\$darwin" && ! "\$nonstop" ; then
       '' | soft) :;; #(
       *)
         # In POSIX sh, ulimit -n is undefined. That's why the result is checked to see if it worked.
-        # shellcheck disable=SC3045
+        # shellcheck disable=SC2039,SC3045
         ulimit -n "\$MAX_FD" ||
             warn "Could not set maximum file descriptor limit to \$MAX_FD"
     esac

